corehq/apps/data_dictionary/views.py

```diff
@@ -403,6 +406,12 @@ def _process_fhir_resource_type_mapping_sheet(domain, worksheet):
             errors.append(_('Not enough columns in {} sheet').format(FHIR_RESOURCE_TYPE_MAPPING_SHEET))
         else:
             case_type, fhir_resource_type = [cell.value for cell in row[:2]]
-            fhir_resource_type_obj = _update_fhir_resource_type(domain, case_type, fhir_resource_type)
-            fhir_resource_type_by_case_type[case_type] = fhir_resource_type_obj
+            try:
+                fhir_resource_type_obj = _update_fhir_resource_type(domain, case_type, fhir_resource_type)
+            except ValidationError as e:
+                for key, msgs in dict(e).items():
+                    for msg in msgs:
+                        errors.append(_("{}: {}").format(key, msg))
```

Comment:
@kaapstorm 
In case i try to leave fhir resource type as blank, django gives me

`name: This field cannot be null.`

because we have the property called "name" on the FHIRResourceType for the resource type, can we rename that to "resource_type" or "fhir_resource" or "fhir_resource_type" instead, so that such messages are better?

Reply:
Having a model named "Foo" with a property named "foo" feels wrong. And FHIRResourceType already has a "case_type" property that is a foreign key, so it might be confusing that "resource_type" is a string. How about "resource_type_name"?

Alternatively, maybe "name" is a reasonable field name, and matches the way types have names generally, like
```python
>>> foo = None
>>> type(foo)
<class 'NoneType'>
>>> type(foo).__name__
'NoneType'
```

What if we just catch "name", and change it to the column name:
```suggestion
                        column = 'Resource type' if key == 'name' else key.capitalize()
                        errors.append(_("{}: {}").format(column, msg))
```
This might make more sense than it would in a UI, because the user knows the column name, not a form field.

Reply:
I see your point, from a model perspective "name" should be enough. 

Let me try a better way to show that message. Immediate thought is to append "Resource Type" in the error messages for FHIRResourceType to make it clear that the message is for the resource type and not case props. Catching the key and replacing it looks like an overkill.